### PR TITLE
SA user bind when enabling auth

### PR DESCRIPTION
**Pull request by mrajashree**
Service account user bind should take place when generating token only if auth
is not enabled.

### CODE REVIEW

**Comment by mrajashree** [comment]
https://github.com/rancher/rancher/issues/9650
Regression caused when removing service account bind for the bug https://github.com/rancher/rancher/issues/9456

File: providers/ldap/ldap_client.go [hunk 1]
```diff
@@ -120,6 +121,21 @@ func (l *LClient) GenerateToken(jsonInput map[string]string) (model.Token, int,
...
+		log.Debugf("Bind service account username password")
+		sausername := getUserExternalID(l.SearchConfig.BindDN, l.Config.LoginDomain)
+		err = lConn.Bind(sausername, l.SearchConfig.BindPassword)
+
+		if err != nil {
+			if ldap.IsErrorWithCode(err, ldap.LDAPResultInvalidCredentials) {
+				status = 401
+			}
+			defer lConn.Close()
+			return nilToken, status, fmt.Errorf("Error %v in ldap bind of service account", err)
```
Comment (prachidamle) [inline providers/ldap/ldap_client.go]: does the error message shown on UI look good?

File: providers/ldap/ldap_client.go [hunk 2]
```diff
@@ -606,7 +622,17 @@ func (l *LClient) TestLogin(testAuthConfig *model.TestAuthConfig) (int, error) {
...
-	log.Info("Binding username password")
+	log.Debugf("Binding service account username password")
+	sausername := getUserExternalID(testAuthConfig.AuthConfig.LdapConfig.ServiceAccountUsername, testAuthConfig.AuthConfig.LdapConfig.LoginDomain)
+	err = lConn.Bind(sausername, testAuthConfig.AuthConfig.LdapConfig.ServiceAccountPassword)
+	if err != nil {
+		if ldap.IsErrorWithCode(err, ldap.LDAPResultInvalidCredentials) {
+			status = 401
+		}
+		return status, fmt.Errorf("Error %v in ldap bind for service account", err)
+	}
```
Comment (prachidamle) [inline providers/ldap/ldap_client.go]: same as above

**Comment by prachidamle** [comment]
LGTM, since the error message look correct on UI.

**Pull request merged**